### DISCUSSION TITLE
Bad performance on a 2017 Model S

### DISCUSSION THREAD

**Discussion by isaac-mcfadyen** [tesla-android/issue-tracker#99] [General]
Hey 👋 

So I deployed Tesla Android 2022.38.1 on my Tesla Model S from 2017 today (pre-MCU upgrade) 🎉 

Unfortunately, it seems that something's severely impacting performance. When the Tesla's browser displays anything at all (sometimes it's a white screen), there is between 30 seconds and 2 minutes between frames. Input also seems to be very delayed, with the time between touching a button and anything taking effect being around 45 seconds.

This occurs both in normal Tesla Android as well as when AutoKit is opened, so I'm guessing it's not an effect of Carplay.

Is this an effect of the very old MCU in these Teslas? The issues aren't occuring when opening Tesla Android on a browser on a PC (very smooth framerate, no issues).

Thanks!

**Comment by mikegapinski** [comment]
MCU 1 is not supported at all, it's too slow and does not ship with a recent browser anymore.
